File: 50 Coding Challenge Part II/JavaScript/Challenge50.js
//  Challenge 50: Create a function that will receive n as argument and return an array of n
// random numbers from 1 to n


function getRandomNumbers(n){
    let arr = []
    for (let i = 1; i <= n; i++){
        arr.push(i);
    }

    shuffleArray(arr);

    return arr;
}


// shuffle method implemented using Fisher-yates shuffle algoeithm
function shuffleArray(arr) {
    let i = arr.length;
    while (--i > 0) {
      let randIndex = Math.floor(Math.random() * (i + 1));
      [arr[randIndex], arr[i]] = [arr[i], arr[randIndex]];
    }
    return arr;
  }

// Driver Method
console.log(getRandomNumbers(10))

File: 50 Coding Challenge Part II/Python/Challenge50.py
# Challenge 50: Create a function that will receive n as argument and return an array of n
# random numbers from 1 to n
import random


# function generates nth random numbers
def generateRandomNumbers(n):
    arr = []
    for num in range(n):
        arr.append(num)

    return shuffleArray(arr)


# function shuffles an array of elements
def shuffleArray(arr):
    n = len(arr) - 1
    for i in range(n):
        random_index = random.randint(0, n)
        temp = arr.pop(random_index)
        arr.append(temp)
    return arr


# Driver Method
def main():
    print(generateRandomNumbers(10))


if __name__ == "__main__":
    main()

